--- koffice-1.5.92/lib/kross/python/cxx/Extensions.hxx.orig	2006-09-20 08:51:10.000000000 +0200
+++ koffice-1.5.92/lib/kross/python/cxx/Extensions.hxx	2006-10-09 21:39:21.056659500 +0200
@@ -425,9 +425,9 @@
 		virtual Object number_power( const Object &, const Object & );
 		
 		// Buffer
-		virtual int buffer_getreadbuffer( int, void** );
-		virtual int buffer_getwritebuffer( int, void** );
-		virtual int buffer_getsegcount( int* );
+		virtual Py_ssize_t buffer_getreadbuffer( Py_ssize_t, void** );
+		virtual Py_ssize_t buffer_getwritebuffer( Py_ssize_t, void** );
+		virtual Py_ssize_t buffer_getsegcount( Py_ssize_t* );
 		
 	private:
 		void missing_method( void );
--- koffice-1.5.92/lib/kross/python/cxx/cxx_extensions.cxx.orig	2006-09-20 08:51:10.000000000 +0200
+++ koffice-1.5.92/lib/kross/python/cxx/cxx_extensions.cxx	2006-10-09 21:40:14.203981000 +0200
@@ -148,15 +148,15 @@
 	static PyObject* call_handler (PyObject*, PyObject*, PyObject*);
 
 	// Sequence methods
-	static int sequence_length_handler(PyObject*);
+	static Py_ssize_t sequence_length_handler(PyObject*);
 	static PyObject* sequence_concat_handler(PyObject*,PyObject*);
-	static PyObject* sequence_repeat_handler(PyObject*, int);
-	static PyObject* sequence_item_handler(PyObject*, int);
-	static PyObject* sequence_slice_handler(PyObject*, int, int);
-	static int sequence_ass_item_handler(PyObject*, int, PyObject*);
-	static int sequence_ass_slice_handler(PyObject*, int, int, PyObject*);
+	static PyObject* sequence_repeat_handler(PyObject*, Py_ssize_t);
+	static PyObject* sequence_item_handler(PyObject*, Py_ssize_t);
+	static PyObject* sequence_slice_handler(PyObject*, Py_ssize_t, Py_ssize_t);
+	static int sequence_ass_item_handler(PyObject*, Py_ssize_t, PyObject*);
+	static int sequence_ass_slice_handler(PyObject*, Py_ssize_t, Py_ssize_t, PyObject*);
 	// Mapping
-	static int mapping_length_handler(PyObject*);
+	static Py_ssize_t mapping_length_handler(PyObject*);
 	static PyObject* mapping_subscript_handler(PyObject*, PyObject*);
 	static int mapping_ass_subscript_handler(PyObject*, PyObject*, PyObject*);
 
@@ -185,9 +185,9 @@
 	static PyObject* number_power_handler(PyObject*, PyObject*, PyObject*);
 
 	// Buffer
-	static int buffer_getreadbuffer_handler (PyObject*, int, void**);
-	static int buffer_getwritebuffer_handler (PyObject*, int, void**);
-	static int buffer_getsegcount_handler (PyObject*, int*);
+	static Py_ssize_t buffer_getreadbuffer_handler (PyObject*, Py_ssize_t, void**);
+	static Py_ssize_t buffer_getwritebuffer_handler (PyObject*, Py_ssize_t, void**);
+	static Py_ssize_t buffer_getsegcount_handler (PyObject*, Py_ssize_t*);
 	}
 
 
@@ -555,7 +555,7 @@
 
 
 // Sequence methods
-extern "C" int sequence_length_handler( PyObject *self )
+extern "C" Py_ssize_t sequence_length_handler( PyObject *self )
 	{
 	try
 		{
@@ -581,7 +581,7 @@
 		}
 	}
 
-extern "C" PyObject* sequence_repeat_handler( PyObject *self, int count )
+extern "C" PyObject* sequence_repeat_handler( PyObject *self, Py_ssize_t count )
 	{
 	try
 		{
@@ -594,7 +594,7 @@
 		}
 	}
 
-extern "C" PyObject* sequence_item_handler( PyObject *self, int index )
+extern "C" PyObject* sequence_item_handler( PyObject *self, Py_ssize_t index )
 	{
 	try
 		{
@@ -607,7 +607,7 @@
 		}
 	}
 
-extern "C" PyObject* sequence_slice_handler( PyObject *self, int first, int last )
+extern "C" PyObject* sequence_slice_handler( PyObject *self, Py_ssize_t first, Py_ssize_t last )
 	{
 	try
 		{
@@ -620,7 +620,7 @@
 		}
 	}
 
-extern "C" int sequence_ass_item_handler( PyObject *self, int index, PyObject *value )
+extern "C" int sequence_ass_item_handler( PyObject *self, Py_ssize_t index, PyObject *value )
 	{
 	try
 		{
@@ -633,7 +633,7 @@
 		}
 	}
 
-extern "C" int sequence_ass_slice_handler( PyObject *self, int first, int last, PyObject *value )
+extern "C" int sequence_ass_slice_handler( PyObject *self, Py_ssize_t first, Py_ssize_t last, PyObject *value )
 	{
 	try
 		{
@@ -647,7 +647,7 @@
 	}
 
 // Mapping
-extern "C" int mapping_length_handler( PyObject *self )
+extern "C" Py_ssize_t mapping_length_handler( PyObject *self )
 	{
 	try
 		{
@@ -974,7 +974,7 @@
 	}
 
 // Buffer
-extern "C" int buffer_getreadbuffer_handler( PyObject *self, int index, void **pp )
+extern "C" Py_ssize_t buffer_getreadbuffer_handler( PyObject *self, Py_ssize_t index, void **pp )
 	{
 	try
 		{
@@ -987,7 +987,7 @@
 		}
 	}
 
-extern "C" int buffer_getwritebuffer_handler( PyObject *self, int index, void **pp )
+extern "C" Py_ssize_t buffer_getwritebuffer_handler( PyObject *self, Py_ssize_t index, void **pp )
 	{
 	try
 		{
@@ -1000,7 +1000,7 @@
 		}
 	}
 
-extern "C" int buffer_getsegcount_handler( PyObject *self, int *count )
+extern "C" Py_ssize_t buffer_getsegcount_handler( PyObject *self, Py_ssize_t *count )
 	{
 	try
 		{
@@ -1162,13 +1162,13 @@
 
 
 // Buffer
-int PythonExtensionBase::buffer_getreadbuffer( int, void** )
+Py_ssize_t PythonExtensionBase::buffer_getreadbuffer( Py_ssize_t, void** )
 	{ missing_method( buffer_getreadbuffer ); return -1; }
 
-int PythonExtensionBase::buffer_getwritebuffer( int, void** )
+Py_ssize_t PythonExtensionBase::buffer_getwritebuffer( Py_ssize_t, void** )
 	{ missing_method( buffer_getwritebuffer ); return -1; }
 
-int PythonExtensionBase::buffer_getsegcount( int* )
+Py_ssize_t PythonExtensionBase::buffer_getsegcount( Py_ssize_t* )
 	{ missing_method( buffer_getsegcount ); return -1; }
 
 //--------------------------------------------------------------------------------
