--- koffice-1.1.1/kchart/kchart.desktop~	Mon Jun 17 18:42:23 2002
+++ koffice-1.1.1/kchart/kchart.desktop	Mon Jun 17 18:42:23 2002
@@ -46,7 +46,7 @@
 Exec=kchart
 X-KDE-Library=libkchartpart
 Type=Application
-Icon=kchart
+Icon=kchart.png
 MimeType=application/x-kchart
 ServiceTypes=KOfficePart,KParts/ReadOnlyPart,KParts/ReadWritePart,KOfficeChart
 X-KDE-NativeMimeType=application/x-kchart
--- koffice-1.1.1/kivio/kiviopart/kivio.desktop~	Mon Jun 17 18:47:01 2002
+++ koffice-1.1.1/kivio/kiviopart/kivio.desktop	Mon Jun 17 18:47:01 2002
@@ -38,6 +38,6 @@
 X-KDE-Library=libkiviopart
 MimeType=application/x-kivio
 Type=Application
-Icon=kivio
+Icon=kivio.png
 ServiceTypes=KOfficePart,KParts/ReadOnlyPart,KParts/ReadWritePart
 X-KDE-NativeMimeType=application/x-kivio
--- koffice-1.1.1/kformula/kformula.desktop~	Mon Jun 17 18:48:02 2002
+++ koffice-1.1.1/kformula/kformula.desktop	Mon Jun 17 18:48:02 2002
@@ -46,6 +46,6 @@
 Comment[zh_CN]=公式编辑器
 MimeType=application/x-kformula
 Type=Application
-Icon=kformula
+Icon=kformula.png
 ServiceTypes=KOfficePart,KParts/ReadOnlyPart,KParts/ReadWritePart
 X-KDE-NativeMimeType=application/x-kformula
--- koffice-1.1.1/kpresenter/kpresenter.desktop~	Mon Jun 17 18:48:57 2002
+++ koffice-1.1.1/kpresenter/kpresenter.desktop	Mon Jun 17 18:48:57 2002
@@ -46,7 +46,7 @@
 Comment[zh_CN]=幻灯片演示
 MimeType=application/x-kpresenter
 Type=Application
-Icon=kpresenter
+Icon=kpresenter.png
 ServiceTypes=KOfficePart,KParts/ReadOnlyPart,KParts/ReadWritePart
 X-KDE-NativeMimeType=application/x-kpresenter
 X-DCOP-ServiceType=multi
--- koffice-1.1.1/kugar/part/kugar.desktop~	Mon Jun 17 18:50:34 2002
+++ koffice-1.1.1/kugar/part/kugar.desktop	Mon Jun 17 18:50:34 2002
@@ -3,7 +3,7 @@
 Name[af]=KUGAR
 Name[eo]=Raportilo
 Exec=kugar
-Icon=kugar
+Icon=kugar.png
 MimeType=application/x-kugar;
 Type=Application
 DocPath=kugar/index.html
--- koffice-1.1.1/kspread/kspread.desktop~	Mon Jun 17 18:51:45 2002
+++ koffice-1.1.1/kspread/kspread.desktop	Mon Jun 17 18:51:45 2002
@@ -42,7 +42,7 @@
 Comment[zh_CN]=电子表格
 MimeType=application/x-kspread
 Type=Application
-Icon=kspread
+Icon=kspread.png
 ServiceTypes=KOfficePart,KParts/ReadOnlyPart,KParts/ReadWritePart,text/x-csv,application/x-kspread-crypt,application/x-quattropro
 X-KDE-NativeMimeType=application/x-kspread
 X-DCOP-ServiceType=multi
--- koffice-1.1.1/kword/kword.desktop~	Mon Jun 17 18:52:30 2002
+++ koffice-1.1.1/kword/kword.desktop	Mon Jun 17 18:52:30 2002
@@ -46,7 +46,7 @@
 Comment[zh_CN]=字处理
 MimeType=application/x-kword
 Type=Application
-Icon=kword
+Icon=kword.png
 ServiceTypes=KOfficePart,KParts/ReadOnlyPart,KParts/ReadWritePart,text/plain,application/x-kword-crypt
 X-KDE-NativeMimeType=application/x-kword
 DocPath=kword/index.html
--- koffice-1.1.1/koshell/koshell.desktop~	Mon Jun 17 18:53:15 2002
+++ koffice-1.1.1/koshell/koshell.desktop	Mon Jun 17 18:53:15 2002
@@ -94,5 +94,5 @@
 Comment[zh_CN]=KOffice 多文档窗口
 Comment[zh_TW]=KOffice 多文件視窗
 Type=Application
-Icon=koshell
+Icon=koshell.png
 X-KDE-StartupNotify=true
