--- koffice-030613/tools/kthesaurus/KThesaurus.desktop~	2003-04-29 01:33:11.000000000 +0200
+++ koffice-030613/tools/kthesaurus/KThesaurus.desktop	2003-06-17 00:18:18.000000000 +0200
@@ -56,4 +56,4 @@
 GenericName[zh_CN]=相关单词
 GenericName[zh_TW]=相關單字
 X-KDE-StartupNotify=true
-Categories=Qt;KDE;Office
+Categories=Qt;KDE;Office;X-Misc;
--- koffice-030613/karbon/data/karbon.desktop~	2003-02-06 14:12:11.000000000 +0100
+++ koffice-030613/karbon/data/karbon.desktop	2003-06-17 00:00:51.000000000 +0200
@@ -51,4 +51,4 @@
 Icon=karbon
 ServiceTypes=KOfficePart,KParts/ReadOnlyPart,KParts/ReadWritePart
 X-KDE-NativeMimeType=application/x-karbon
-Categories=Qt;KDE;Office
+Categories=Qt;KDE;Graphics;
--- koffice-030613/kchart/kchart.desktop~	2003-04-29 01:30:54.000000000 +0200
+++ koffice-030613/kchart/kchart.desktop	2003-06-17 00:00:35.000000000 +0200
@@ -59,4 +59,4 @@
 X-KDE-NativeMimeType=application/x-kchart
 X-KDE-StartupNotify=true
 X-DCOP-ServiceType=Multi
-Categories=Qt;KDE;Office
+Categories=Qt;KDE;Office;X-Misc;
--- koffice-030613/kexi/kexi.desktop~	2003-05-28 01:16:30.000000000 +0200
+++ koffice-030613/kexi/kexi.desktop	2003-06-17 00:02:55.000000000 +0200
@@ -51,4 +51,4 @@
 X-KDE-NativeMimeType=application/x-vnd.kde.kexi
 X-KDE-Library=libkexipart
 X-KDE-NOTKoDocumentEmbeddable=true
-Categories=Qt;KDE;Office
+Categories=Qt;KDE;Office;X-Database
--- koffice-030613/kformula/kformula.desktop~	2003-04-29 01:31:20.000000000 +0200
+++ koffice-030613/kformula/kformula.desktop	2003-06-17 00:03:45.000000000 +0200
@@ -68,4 +68,4 @@
 X-KDE-NativeMimeType=application/x-kformula
 X-DCOP-ServiceType=Multi
 DocPath=kformula/index.html
-Categories=Qt;KDE;Office
+Categories=Qt;KDE;Office;X-Misc;
--- koffice-030613/kivio/kiviopart/kivio.desktop~	2003-04-29 01:31:23.000000000 +0200
+++ koffice-030613/kivio/kiviopart/kivio.desktop	2003-06-17 00:04:57.000000000 +0200
@@ -42,4 +42,4 @@
 ServiceTypes=KOfficePart,KParts/ReadOnlyPart,KParts/ReadWritePart
 X-KDE-NativeMimeType=application/x-kivio
 X-DCOP-ServiceType=Multi
-Categories=Qt;KDE;Office
+Categories=Qt;KDE;Office;X-Misc;
--- koffice-030613/kpresenter/kpresenter.desktop~	2003-04-29 01:32:01.000000000 +0200
+++ koffice-030613/kpresenter/kpresenter.desktop	2003-06-17 00:07:45.000000000 +0200
@@ -67,4 +67,4 @@
 X-DCOP-ServiceType=multi
 DocPath=kpresenter/index.html
 X-KDE-StartupNotify=true
-Categories=Qt;KDE;Office
+Categories=Qt;KDE;Office;Presentation;
--- koffice-030613/krita/krita.desktop~	2003-05-08 01:20:06.000000000 +0200
+++ koffice-030613/krita/krita.desktop	2003-06-17 00:08:53.000000000 +0200
@@ -60,3 +60,4 @@
 X-KDE-NativeMimeType=application/x-krita
 X-KDE-StartupNotify=true
 X-DCOP-ServiceType=Multi
+Categories=Qt;KDE;Graphics;
--- koffice-030613/kspread/kspread.desktop~	2003-04-29 01:32:21.000000000 +0200
+++ koffice-030613/kspread/kspread.desktop	2003-06-17 00:16:43.000000000 +0200
@@ -63,4 +63,4 @@
 X-DCOP-ServiceType=multi
 DocPath=kspread/index.html
 X-KDE-StartupNotify=true
-Categories=Qt;KDE;Office
+Categories=Qt;KDE;Office;Spreadsheet;
--- koffice-030613/kugar/kudesigner/kudesigner.desktop~	2003-06-02 01:23:33.000000000 +0200
+++ koffice-030613/kugar/kudesigner/kudesigner.desktop	2003-06-17 00:13:47.000000000 +0200
@@ -103,5 +103,4 @@
 X-KDE-NativeMimeType=application/x-kudesigner
 X-KDE-Library=libkudesignerpart
 ServiceTypes=KOfficePart,KParts/ReadOnlyPart,KParts/ReadWritePart
-
-Categories=Qt;KDE;Office
+Categories=Qt;KDE;Office;X-Misc;
--- koffice-030613/kugar/part/kugar.desktop~	2003-04-29 01:32:36.000000000 +0200
+++ koffice-030613/kugar/part/kugar.desktop	2003-06-17 00:14:21.000000000 +0200
@@ -58,4 +58,4 @@
 ServiceTypes=KOfficePart,KParts/ReadOnlyPart,Browser/View
 X-KDE-Library=libkugarpart
 X-KDE-NativeMimeType=application/x-vnd.kde.kugar.mixed
-Categories=Qt;KDE;Office
+Categories=Qt;KDE;Office;X-Misc;
--- koffice-030613/kword/kword.desktop~	2003-04-29 01:32:41.000000000 +0200
+++ koffice-030613/kword/kword.desktop	2003-06-17 00:15:43.000000000 +0200
@@ -65,4 +65,4 @@
 DocPath=kword/index.html
 X-KDE-StartupNotify=true
 X-DCOP-ServiceType=Multi
-Categories=Qt;KDE;Office
+Categories=Qt;KDE;Office;WordProcessor;
--- koffice-030613/kexi/formeditor/test/kformdesigner.desktop~	2003-06-17 02:46:15.000000000 +0200
+++ koffice-030613/kexi/formeditor/test/kformdesigner.desktop	2003-06-17 02:45:19.000000000 +0200
@@ -13,3 +13,4 @@
 Comment[pt_BR]=Um aplicativo KDE KPart
 Comment[sv]=Ett KDE Kpart-program
 Terminal=0
+Categories=Qt;KDE;Office;X-Misc;
