Index: acinclude.m4.in
===================================================================
RCS file: /home/kde/kde-common/admin/acinclude.m4.in,v
retrieving revision 2.440
retrieving revision 2.432
diff -u -r2.440 -r2.432
--- admin/acinclude.m4.in~	25 Jan 2004 10:51:23 -0000	2.440
+++ admin/acinclude.m4.in	5 Dec 2003 12:18:43 -0000	2.432
@@ -29,7 +29,7 @@
 # KDE_PATH_X_DIRECT
 dnl Internal subroutine of AC_PATH_X.
 dnl Set ac_x_includes and/or ac_x_libraries.
-AC_DEFUN([KDE_PATH_X_DIRECT],
+AC_DEFUN(KDE_PATH_X_DIRECT,
 [
 AC_REQUIRE([KDE_CHECK_LIB64])
 
@@ -156,7 +156,7 @@
 dnl Find a file (or one of more files in a list of dirs)
 dnl ------------------------------------------------------------------------
 dnl
-AC_DEFUN([AC_FIND_FILE],
+AC_DEFUN(AC_FIND_FILE,
 [
 $3=NO
 for i in $2;
@@ -175,7 +175,7 @@
 
 dnl KDE_FIND_PATH(programm-name, variable-name, list of directories,
 dnl	if-not-found, test-parameter)
-AC_DEFUN([KDE_FIND_PATH],
+AC_DEFUN(KDE_FIND_PATH,
 [
    AC_MSG_CHECKING([for $1])
    if test -n "$$2"; then
@@ -228,7 +228,7 @@
    fi
 ])
 
-AC_DEFUN([KDE_MOC_ERROR_MESSAGE],
+AC_DEFUN(KDE_MOC_ERROR_MESSAGE,
 [
     AC_MSG_ERROR([No Qt meta object compiler (moc) found!
 Please check whether you installed Qt correctly.
@@ -240,7 +240,7 @@
 ])
 ])
 
-AC_DEFUN([KDE_UIC_ERROR_MESSAGE],
+AC_DEFUN(KDE_UIC_ERROR_MESSAGE,
 [
     AC_MSG_WARN([No Qt ui compiler (uic) found!
 Please check whether you installed Qt correctly.
@@ -253,7 +253,7 @@
 ])
 
 
-AC_DEFUN([KDE_CHECK_UIC_FLAG],
+AC_DEFUN(KDE_CHECK_UIC_FLAG,
 [
     AC_MSG_CHECKING([whether uic supports -$1 ])
     kde_cache=`echo $1 | sed 'y% .=/+-%____p_%'`
@@ -288,7 +288,7 @@
 dnl in $QTDIR/bin, and some more usual places
 dnl ------------------------------------------------------------------------
 dnl
-AC_DEFUN([AC_PATH_QT_MOC_UIC],
+AC_DEFUN(AC_PATH_QT_MOC_UIC,
 [
    AC_REQUIRE([KDE_CHECK_PERL])
    qt_bindirs=""
@@ -336,7 +336,7 @@
    AC_SUBST(UIC_TR)
 ])
 
-AC_DEFUN([KDE_1_CHECK_PATHS],
+AC_DEFUN(KDE_1_CHECK_PATHS,
 [
   KDE_1_CHECK_PATH_HEADERS
 
@@ -384,7 +384,7 @@
 
 ])
 
-AC_DEFUN([KDE_SET_PATHS],
+AC_DEFUN(KDE_SET_PATHS,
 [
   kde_cv_all_paths="kde_have_all_paths=\"yes\" \
 	kde_htmldir=\"$kde_htmldir\" \
@@ -412,7 +412,7 @@
 	kde_result=$1"
 ])
 
-AC_DEFUN([KDE_SET_DEFAULT_PATHS],
+AC_DEFUN(KDE_SET_DEFAULT_PATHS,
 [
 if test "$1" = "default"; then
 
@@ -501,7 +501,7 @@
 fi
 ])
 
-AC_DEFUN([KDE_CHECK_PATHS_FOR_COMPLETENESS],
+AC_DEFUN(KDE_CHECK_PATHS_FOR_COMPLETENESS,
 [ if test -z "$kde_htmldir" || test -z "$kde_appsdir" ||
    test -z "$kde_icondir" || test -z "$kde_sounddir" ||
    test -z "$kde_datadir" || test -z "$kde_locale"  ||
@@ -518,14 +518,14 @@
   fi
 ])
 
-AC_DEFUN([KDE_MISSING_PROG_ERROR],
+AC_DEFUN(KDE_MISSING_PROG_ERROR,
 [
     AC_MSG_ERROR([The important program $1 was not found!
 Please check whether you installed KDE correctly.
 ])
 ])
 
-AC_DEFUN([KDE_MISSING_ARTS_ERROR],
+AC_DEFUN(KDE_MISSING_ARTS_ERROR,
 [
     AC_MSG_ERROR([The important program $1 was not found!
 Please check whether you installed aRts correctly or use
@@ -533,10 +533,10 @@
 ])
 ])
 
-AC_DEFUN([KDE_SUBST_PROGRAMS],
+AC_DEFUN(KDE_SUBST_PROGRAMS,
 [
     AC_ARG_WITH(arts,
-        AC_HELP_STRING([--without-arts],[build without aRts [default=yes]]),
+        [  --without-arts        build without aRts [default=yes] ],
         [build_arts=$withval],
         [build_arts=yes]
     )
@@ -623,7 +623,7 @@
         AC_SUBST(kde_libs_htmldir)
 ])dnl
 
-AC_DEFUN([AC_CREATE_KFSSTND],
+AC_DEFUN(AC_CREATE_KFSSTND,
 [
 AC_REQUIRE([AC_CHECK_RPATH])
 
@@ -671,7 +671,7 @@
 
 ])
 
-AC_DEFUN([AC_SUBST_KFSSTND],
+AC_DEFUN(AC_SUBST_KFSSTND,
 [
 AC_SUBST(kde_htmldir)
 AC_SUBST(kde_appsdir)
@@ -708,7 +708,7 @@
 dnl AC_SUBST(kde_toolbardir)
 ])
 
-AC_DEFUN([KDE_MISC_TESTS],
+AC_DEFUN(KDE_MISC_TESTS,
 [
    dnl Checks for libraries.
    AC_CHECK_LIB(util, main, [LIBUTIL="-lutil"]) dnl for *BSD 
@@ -843,28 +843,28 @@
 dnl macro AC_PATH_X
 dnl ------------------------------------------------------------------------
 dnl
-AC_DEFUN([K_PATH_X],
+AC_DEFUN(K_PATH_X,
 [
 AC_REQUIRE([KDE_MISC_TESTS])dnl
 AC_REQUIRE([KDE_CHECK_LIB64])
 
 AC_ARG_ENABLE(
   embedded,
-  AC_HELP_STRING([--enable-embedded],[link to Qt-embedded, don't use X]),
+  [  --enable-embedded       link to Qt-embedded, don't use X],
   kde_use_qt_emb=$enableval,
   kde_use_qt_emb=no
 )
 
 AC_ARG_ENABLE(
   qtopia,
-  AC_HELP_STRING([--enable-qtopia],[link to Qt-embedded, link to the Qtopia Environment]),
+  [  --enable-qtopia         link to Qt-embedded, link to the Qtopia Environment],
   kde_use_qt_emb_palm=$enableval,
   kde_use_qt_emb_palm=no
 )
 
 AC_ARG_ENABLE(
   mac,
-  AC_HELP_STRING([--enable-mac],[link to Qt/Mac (don't use X)]),
+  [  --enable-mac       link to Qt/Mac (don't use X)],
   kde_use_qt_mac=$enableval,
   kde_use_qt_mac=no
 )
@@ -1060,7 +1060,7 @@
 
 ])
 
-AC_DEFUN([KDE_PRINT_QT_PROGRAM],
+AC_DEFUN(KDE_PRINT_QT_PROGRAM,
 [
 AC_REQUIRE([KDE_USE_QT])
 cat > conftest.$ac_ext <<EOF
@@ -1123,7 +1123,7 @@
 EOF
 ])
 
-AC_DEFUN([KDE_USE_QT],
+AC_DEFUN(KDE_USE_QT,
 [
 if test -z "$1"; then
   # Current default Qt version: 3.2
@@ -1164,7 +1164,7 @@
     kde_qt_minversion=">= 1.42 and < 2.0"
   fi
 else
-   kde_qt_minversion="$2"
+   kde_qt_minversion=$2
 fi
 
 if test -z "$3"; then
@@ -1188,7 +1188,7 @@
     kde_qt_verstring="QT_VERSION >= 142 && QT_VERSION < 200"
    fi
 else
-   kde_qt_verstring="$3"
+   kde_qt_verstring=$3
 fi
 
 if test $kde_qtver = 3; then
@@ -1202,7 +1202,7 @@
 fi
 ])
 
-AC_DEFUN([KDE_CHECK_QT_DIRECT],
+AC_DEFUN(KDE_CHECK_QT_DIRECT,
 [
 AC_REQUIRE([KDE_USE_QT])
 AC_MSG_CHECKING([if Qt compiles without flags])
@@ -1265,7 +1265,7 @@
 dnl and $(QT_INCLUDES) will be -Iqthdrlocation (if needed)
 dnl ------------------------------------------------------------------------
 dnl
-AC_DEFUN([AC_PATH_QT_1_3],
+AC_DEFUN(AC_PATH_QT_1_3,
 [
 AC_REQUIRE([K_PATH_X])
 AC_REQUIRE([KDE_USE_QT])
@@ -1277,7 +1277,7 @@
 
 AC_ARG_ENABLE(
   mt,
-  AC_HELP_STRING([--disable-mt],[link to non-threaded Qt (deprecated)]),
+  [  --disable-mt            link to non-threaded Qt (deprecated)],
   kde_use_qt_mt=$enableval,
   [
     if test $kde_qtver = 3; then
@@ -1392,14 +1392,14 @@
 qt_libraries=""
 qt_includes=""
 AC_ARG_WITH(qt-dir,
-    AC_HELP_STRING([--with-qt-dir=DIR],[where the root of Qt is installed ]),
+    [  --with-qt-dir=DIR       where the root of Qt is installed ],
     [  ac_qt_includes="$withval"/include
        ac_qt_libraries="$withval"/lib${kdelibsuff}
        ac_qt_bindir="$withval"/bin
     ])
 
 AC_ARG_WITH(qt-includes,
-    AC_HELP_STRING([--with-qt-includes=DIR],[where the Qt includes are. ]),
+    [  --with-qt-includes=DIR  where the Qt includes are. ],
     [
        ac_qt_includes="$withval"
     ])
@@ -1407,7 +1407,7 @@
 kde_qt_libs_given=no
 
 AC_ARG_WITH(qt-libraries,
-    AC_HELP_STRING([--with-qt-libraries=DIR],[where the Qt library is installed.]),
+    [  --with-qt-libraries=DIR where the Qt library is installed.],
     [  ac_qt_libraries="$withval"
        kde_qt_libs_given=yes
     ])
@@ -1571,12 +1571,12 @@
 AC_SUBST(kde_qtver)
 ])
 
-AC_DEFUN([AC_PATH_QT],
+AC_DEFUN(AC_PATH_QT,
 [
 AC_PATH_QT_1_3
 ])
 
-AC_DEFUN([KDE_CHECK_UIC_PLUGINS],
+AC_DEFUN(KDE_CHECK_UIC_PLUGINS,
 [
 AC_REQUIRE([AC_PATH_QT_MOC_UIC])
 
@@ -1624,11 +1624,9 @@
 fi
 ])
 
-AC_DEFUN([KDE_CHECK_FINAL],
+AC_DEFUN(KDE_CHECK_FINAL,
 [
-  AC_ARG_ENABLE(final,
-	AC_HELP_STRING([--enable-final],
-		       [build size optimized apps (experimental - needs lots of memory)]),
+  AC_ARG_ENABLE(final, [  --enable-final          build size optimized apps (experimental - needs lots of memory)],
 	kde_use_final=$enableval, kde_use_final=no)
 
   if test "x$kde_use_final" = "xyes"; then
@@ -1642,10 +1640,9 @@
   AC_SUBST(KDE_USE_FINAL_FALSE)
 ])
 
-AC_DEFUN([KDE_CHECK_CLOSURE],
+AC_DEFUN(KDE_CHECK_CLOSURE,
 [
-  AC_ARG_ENABLE(closure,
-		AC_HELP_STRING([--enable-closure],[delay template instantiation]),
+  AC_ARG_ENABLE(closure, [  --enable-closure       delay template instantiation],
   	kde_use_closure=$enableval, kde_use_closure=no)
 
   KDE_NO_UNDEFINED=""
@@ -1672,9 +1669,9 @@
   AC_SUBST(KDE_NO_UNDEFINED)
 ])
 
-AC_DEFUN([KDE_CHECK_NMCHECK],
+AC_DEFUN(KDE_CHECK_NMCHECK,
 [
-  AC_ARG_ENABLE(nmcheck,AC_HELP_STRING([--enable-nmcheck],[enable automatic namespace cleanness check]),
+  AC_ARG_ENABLE(nmcheck, [  --enable-nmcheck        enable automatic namespace cleanness check],
 	kde_use_nmcheck=$enableval, kde_use_nmcheck=no)
 
   if test "$kde_use_nmcheck" = "yes"; then
@@ -1702,8 +1699,9 @@
 dnl and $(kde_includes) will be the kdehdrlocation (if needed)
 dnl ------------------------------------------------------------------------
 dnl
-AC_DEFUN([AC_BASE_PATH_KDE],
+AC_DEFUN(AC_BASE_PATH_KDE,
 [
+AC_PREREQ([2.13])
 AC_REQUIRE([KDE_CHECK_STL])
 AC_REQUIRE([AC_PATH_QT])dnl
 AC_REQUIRE([KDE_CHECK_LIB64])
@@ -1856,10 +1854,11 @@
 
 ])
 
-AC_DEFUN([KDE_CHECK_EXTRA_LIBS],
+AC_DEFUN(KDE_CHECK_EXTRA_LIBS,
 [
 AC_MSG_CHECKING(for extra includes)
-AC_ARG_WITH(extra-includes,AC_HELP_STRING([--with-extra-includes=DIR],[adds non standard include paths]),
+AC_ARG_WITH(extra-includes, [  --with-extra-includes=DIR
+                          adds non standard include paths],
   kde_use_extra_includes="$withval",
   kde_use_extra_includes=NONE
 )
@@ -1884,7 +1883,7 @@
 
 kde_extra_libs=
 AC_MSG_CHECKING(for extra libs)
-AC_ARG_WITH(extra-libs,AC_HELP_STRING([--with-extra-libs=DIR],[adds non standard library paths]),
+AC_ARG_WITH(extra-libs, [  --with-extra-libs=DIR   adds non standard library paths],
   kde_use_extra_libs=$withval,
   kde_use_extra_libs=NONE
 )
@@ -1910,7 +1909,7 @@
 
 ])
 
-AC_DEFUN([KDE_1_CHECK_PATH_HEADERS],
+AC_DEFUN(KDE_1_CHECK_PATH_HEADERS,
 [
     AC_MSG_CHECKING([for KDE headers installed])
     AC_LANG_SAVE
@@ -1965,7 +1964,7 @@
   AC_LANG_RESTORE
 ])
 
-AC_DEFUN([KDE_CHECK_KDEQTADDON],
+AC_DEFUN(KDE_CHECK_KDEQTADDON,
 [
 AC_MSG_CHECKING(for kde-qt-addon)
 AC_CACHE_VAL(kde_cv_have_kdeqtaddon,
@@ -2001,7 +2000,38 @@
 fi
 ])
 
-AC_DEFUN([KDE_CREATE_LIBS_ALIASES],
+AC_DEFUN(KDE_CHECK_KIMGIO,
+[
+   AC_REQUIRE([AC_BASE_PATH_KDE])
+   AC_REQUIRE([KDE_CHECK_EXTRA_LIBS])
+   AC_REQUIRE([AC_FIND_TIFF])
+   AC_REQUIRE([AC_FIND_JPEG])
+   AC_REQUIRE([AC_FIND_PNG])
+   AC_REQUIRE([AC_FIND_JASPER])
+   AC_REQUIRE([KDE_CREATE_LIBS_ALIASES])
+
+   if test "$1" = "existence"; then
+     AC_LANG_SAVE
+     AC_LANG_CPLUSPLUS
+     kde_save_LIBS="$LIBS"
+     LIBS="$LIBS $all_libraries $LIBJPEG $LIBTIFF $LIBPNG $LIBQT -lm"
+     AC_CHECK_LIB(kimgio, kimgioRegister, [
+      LIBKIMGIO_EXISTS=yes],LIBKIMGIO_EXISTS=no)
+     LIBS="$kde_save_LIBS"
+     AC_LANG_RESTORE
+   else
+     LIBKIMGIO_EXISTS=yes
+   fi
+
+   if test "$LIBKIMGIO_EXISTS" = "yes"; then
+     LIB_KIMGIO='-lkimgio'
+   else
+     LIB_KIMGIO=''
+   fi
+   AC_SUBST(LIB_KIMGIO)
+])
+
+AC_DEFUN(KDE_CREATE_LIBS_ALIASES,
 [
    AC_REQUIRE([KDE_MISC_TESTS])
    AC_REQUIRE([KDE_CHECK_LIBDL])
@@ -2044,10 +2074,10 @@
 fi
 ])
 
-AC_DEFUN([AC_PATH_KDE],
+AC_DEFUN(AC_PATH_KDE,
 [
   AC_BASE_PATH_KDE
-  AC_ARG_ENABLE(path-check,AC_HELP_STRING([--disable-path-check],[don't try to find out, where to install]),
+  AC_ARG_ENABLE(path-check, [  --disable-path-check    don't try to find out, where to install],
   [
   if test "$enableval" = "no";
     then ac_use_path_checking="default"
@@ -2069,7 +2099,7 @@
 ])
 
 dnl KDE_CHECK_FUNC_EXT(<func>, [headers], [sample-use], [C prototype], [autoheader define], [call if found])
-AC_DEFUN([KDE_CHECK_FUNC_EXT],
+AC_DEFUN(KDE_CHECK_FUNC_EXT,
 [
 AC_MSG_CHECKING(for $1)
 AC_CACHE_VAL(kde_cv_func_$1,
@@ -2160,7 +2190,7 @@
 ])
 ])
 
-AC_DEFUN([AC_CHECK_SETENV],
+AC_DEFUN(AC_CHECK_SETENV,
 [
 	KDE_CHECK_FUNC_EXT(setenv, [
 #include <stdlib.h>
@@ -2170,7 +2200,7 @@
 		[SETENV])
 ])
 
-AC_DEFUN([AC_CHECK_UNSETENV],
+AC_DEFUN(AC_CHECK_UNSETENV,
 [
 	KDE_CHECK_FUNC_EXT(unsetenv, [
 #include <stdlib.h>
@@ -2180,7 +2210,7 @@
 		[UNSETENV])
 ])
 
-AC_DEFUN([AC_CHECK_GETDOMAINNAME],
+AC_DEFUN(AC_CHECK_GETDOMAINNAME,
 [
 	KDE_CHECK_FUNC_EXT(getdomainname, [
 #include <stdlib.h>
@@ -2196,7 +2226,7 @@
 		[GETDOMAINNAME])
 ])
 
-AC_DEFUN([AC_CHECK_GETHOSTNAME],
+AC_DEFUN(AC_CHECK_GETHOSTNAME,
 [
 	KDE_CHECK_FUNC_EXT(gethostname, [
 #include <stdlib.h>
@@ -2210,7 +2240,7 @@
 		[GETHOSTNAME])
 ])
 
-AC_DEFUN([AC_CHECK_USLEEP],
+AC_DEFUN(AC_CHECK_USLEEP,
 [
 	KDE_CHECK_FUNC_EXT(usleep, [
 #include <unistd.h>
@@ -2223,7 +2253,7 @@
 ])
 
 
-AC_DEFUN([AC_CHECK_RANDOM],
+AC_DEFUN(AC_CHECK_RANDOM,
 [
 	KDE_CHECK_FUNC_EXT(random, [
 #include <stdlib.h>
@@ -2245,7 +2275,7 @@
 
 ])
 
-AC_DEFUN([AC_CHECK_INITGROUPS],
+AC_DEFUN(AC_CHECK_INITGROUPS,
 [
 	KDE_CHECK_FUNC_EXT(initgroups, [
 #include <sys/types.h>
@@ -2260,7 +2290,7 @@
 	[INITGROUPS])
 ])
 
-AC_DEFUN([AC_CHECK_MKSTEMPS],
+AC_DEFUN(AC_CHECK_MKSTEMPS,
 [
 	KDE_CHECK_FUNC_EXT(mkstemps, [
 #include <stdlib.h>
@@ -2273,7 +2303,7 @@
 	[MKSTEMPS])
 ])
 
-AC_DEFUN([AC_CHECK_MKDTEMP],
+AC_DEFUN(AC_CHECK_MKDTEMP,
 [
 	KDE_CHECK_FUNC_EXT(mkdtemp, [
 #include <stdlib.h>
@@ -2287,7 +2317,7 @@
 ])
 
 
-AC_DEFUN([AC_CHECK_RES_INIT],
+AC_DEFUN(AC_CHECK_RES_INIT,
 [
   AC_MSG_CHECKING([if res_init needs -lresolv])
   kde_libs_safe="$LIBS"
@@ -2331,7 +2361,7 @@
   )
 ])
 
-AC_DEFUN([AC_CHECK_STRLCPY],
+AC_DEFUN(AC_CHECK_STRLCPY,
 [
 	KDE_CHECK_FUNC_EXT(strlcpy, [
 #include <string.h>
@@ -2343,7 +2373,7 @@
 	[STRLCPY])
 ])
 
-AC_DEFUN([AC_CHECK_STRLCAT],
+AC_DEFUN(AC_CHECK_STRLCAT,
 [
 	KDE_CHECK_FUNC_EXT(strlcat, [
 #include <string.h>
@@ -2356,7 +2386,7 @@
 	[STRLCAT])
 ])
 
-AC_DEFUN([AC_FIND_GIF],
+AC_DEFUN(AC_FIND_GIF,
    [AC_MSG_CHECKING([for giflib])
 AC_CACHE_VAL(ac_cv_lib_gif,
 [ac_save_LIBS="$LIBS"
@@ -2390,7 +2420,7 @@
 fi
 ])
 
-AC_DEFUN([KDE_FIND_JPEG_HELPER],
+AC_DEFUN(KDE_FIND_JPEG_HELPER,
 [
 AC_MSG_CHECKING([for libjpeg$2])
 AC_CACHE_VAL(ac_cv_lib_jpeg_$1,
@@ -2432,7 +2462,7 @@
 
 ])
 
-AC_DEFUN([AC_FIND_JPEG],
+AC_DEFUN(AC_FIND_JPEG,
 [
 dnl first look for libraries
 KDE_FIND_JPEG_HELPER(6b, 6b,
@@ -2484,7 +2514,7 @@
 ])
 ])
 
-AC_DEFUN([KDE_CHECK_QT_JPEG],
+AC_DEFUN(KDE_CHECK_QT_JPEG,
 [
 if test -n "$LIBJPEG"; then
 AC_MSG_CHECKING([if Qt needs $LIBJPEG])
@@ -2521,7 +2551,7 @@
 
 ])
 
-AC_DEFUN([AC_FIND_ZLIB],
+AC_DEFUN(AC_FIND_ZLIB,
 [
 AC_REQUIRE([KDE_CHECK_EXTRA_LIBS])
 AC_MSG_CHECKING([for libz])
@@ -2563,7 +2593,7 @@
 AC_SUBST(LIBZ)
 ])
 
-AC_DEFUN([KDE_TRY_TIFFLIB],
+AC_DEFUN(KDE_TRY_TIFFLIB,
 [
 AC_MSG_CHECKING([for libtiff $1])
 
@@ -2609,7 +2639,7 @@
 
 ])
 
-AC_DEFUN([AC_FIND_TIFF],
+AC_DEFUN(AC_FIND_TIFF,
 [
 AC_REQUIRE([K_PATH_X])
 AC_REQUIRE([AC_FIND_ZLIB])
@@ -2623,7 +2653,7 @@
 ])
 
 
-AC_DEFUN([AC_FIND_PNG],
+AC_DEFUN(AC_FIND_PNG,
 [
 AC_REQUIRE([KDE_CHECK_EXTRA_LIBS])
 AC_REQUIRE([AC_FIND_ZLIB])
@@ -2667,7 +2697,7 @@
 ])
 
 
-AC_DEFUN([AC_FIND_JASPER],
+AC_DEFUN(AC_FIND_JASPER,
 [
 AC_REQUIRE([KDE_CHECK_EXTRA_LIBS])
 AC_REQUIRE([AC_FIND_JPEG])
@@ -2703,12 +2733,12 @@
 AC_SUBST(LIB_JASPER)
 ])
 
-AC_DEFUN([AC_CHECK_BOOL],
+AC_DEFUN(AC_CHECK_BOOL,
 [
   AC_DEFINE_UNQUOTED(HAVE_BOOL, 1, [You _must_ have bool])
 ])
 
-AC_DEFUN([AC_CHECK_GNU_EXTENSIONS],
+AC_DEFUN(AC_CHECK_GNU_EXTENSIONS,
 [
 AC_MSG_CHECKING(if you need GNU extensions)
 AC_CACHE_VAL(ac_cv_gnu_extensions,
@@ -2736,7 +2766,7 @@
 fi
 ])
 
-AC_DEFUN([KDE_CHECK_COMPILER_FLAG],
+AC_DEFUN(KDE_CHECK_COMPILER_FLAG,
 [
 AC_MSG_CHECKING([whether $CXX supports -$1])
 kde_cache=`echo $1 | sed 'y% .=/+-,%____p__%'`
@@ -2764,7 +2794,7 @@
 dnl AC_REMOVE_FORBIDDEN removes forbidden arguments from variables
 dnl use: AC_REMOVE_FORBIDDEN(CC, [-forbid -bad-option whatever])
 dnl it's all white-space separated
-AC_DEFUN([AC_REMOVE_FORBIDDEN],
+AC_DEFUN(AC_REMOVE_FORBIDDEN,
 [ __val=$$1
   __forbid=" $2 "
   if test -n "$__val"; then
@@ -2785,7 +2815,7 @@
 ])
 
 dnl AC_VALIDIFY_CXXFLAGS checks for forbidden flags the user may have given
-AC_DEFUN([AC_VALIDIFY_CXXFLAGS],
+AC_DEFUN(AC_VALIDIFY_CXXFLAGS,
 [dnl
 if test "x$kde_use_qt_emb" != "xyes"; then
  AC_REMOVE_FORBIDDEN(CXX, [-fno-rtti -rpath])
@@ -2796,10 +2826,9 @@
 fi
 ])
 
-AC_DEFUN([AC_CHECK_COMPILERS],
+AC_DEFUN(AC_CHECK_COMPILERS,
 [
-  AC_ARG_ENABLE(debug,
-	        AC_HELP_STRING([--enable-debug=ARG],[enables debug symbols (yes|no|full) [default=no]]),
+  AC_ARG_ENABLE(debug,[  --enable-debug[=ARG]    enables debug symbols (yes|no|full) [default=no]],
   [
     case $enableval in
       yes)
@@ -2821,14 +2850,9 @@
   ])
 
   dnl Just for configure --help
-  AC_ARG_ENABLE(dummyoption,
-	        AC_HELP_STRING([--disable-debug],
-	  		       [disables debug output and debug symbols [default=no]]),
-		[],[])
-
-  AC_ARG_ENABLE(strict,
-		AC_HELP_STRING([--enable-strict],
-			      [compiles with strict compiler options (may not work!)]),
+  AC_ARG_ENABLE(dummyoption,[  --disable-debug         disables debug output and debug symbols [default=no]],[],[])
+
+  AC_ARG_ENABLE(strict,[  --enable-strict         compiles with strict compiler options (may not work!)],
    [
     if test $enableval = "no"; then
          kde_use_strict_options="no"
@@ -2837,7 +2861,7 @@
     fi
    ], [kde_use_strict_options="no"])
 
-  AC_ARG_ENABLE(warnings,AC_HELP_STRING([--disable-warnings],[disables compilation with -Wall and similiar]),
+  AC_ARG_ENABLE(warnings,[  --disable-warnings      disables compilation with -Wall and similiar],
    [
     if test $enableval = "no"; then
          kde_use_warnings="no"
@@ -2851,7 +2875,7 @@
     kde_use_warnings=yes
   fi
 
-  AC_ARG_ENABLE(profile,AC_HELP_STRING([--enable-profile],[creates profiling infos [default=no]]),
+  AC_ARG_ENABLE(profile,[  --enable-profile        creates profiling infos [default=no]],
     [kde_use_profiling=$enableval],
     [kde_use_profiling="no"]
   )
@@ -2962,9 +2986,7 @@
     dnl KDE_CHECK_COMPILER_FLAG(-no_exceptions,[CXXFLAGS="$CXXFLAGS --no_exceptions"])
     dnl KDE_CHECK_COMPILER_FLAG(-exceptions, [USE_EXCEPTIONS="--exceptions"], USE_EXCEPTIONS=	)
 
-    AC_ARG_ENABLE(pch,
-	AC_HELP_STRING([--enable-pch],
-		       [enables precompiled header support (currently only KCC) [default=no]]),
+    AC_ARG_ENABLE(pch,[  --enable-pch            enables precompiled header support (currently only KCC) [default=no]],
     [
       kde_use_pch=$enableval
     ],[kde_use_pch=no])
@@ -3033,14 +3055,14 @@
   ifdef([AM_DEPENDENCIES], AC_REQUIRE([KDE_ADD_DEPENDENCIES]), [])
 ])
 
-AC_DEFUN([KDE_ADD_DEPENDENCIES],
+AC_DEFUN(KDE_ADD_DEPENDENCIES,
 [
    [A]M_DEPENDENCIES(CC)
    [A]M_DEPENDENCIES(CXX)
 ])
 
 dnl just a wrapper to clean up configure.in
-AC_DEFUN([KDE_PROG_LIBTOOL],
+AC_DEFUN(KDE_PROG_LIBTOOL,
 [
 AC_REQUIRE([AC_CHECK_COMPILERS])
 AC_REQUIRE([AC_ENABLE_SHARED])
@@ -3064,7 +3086,7 @@
 AC_SUBST(AUTOCONF,'$(SHELL) $(top_srcdir)/admin/cvs.sh configure || touch configure')
 ])
 
-AC_DEFUN([KDE_CHECK_LIB64],
+AC_DEFUN(KDE_CHECK_LIB64,
 [
     kdelibsuff=no
     AC_ARG_ENABLE(libsuffix,
@@ -3089,14 +3111,13 @@
     fi
 ])
 
-AC_DEFUN([KDE_CHECK_TYPES],
+AC_DEFUN(KDE_CHECK_TYPES,
 [  AC_CHECK_SIZEOF(int, 4)dnl
-   AC_CHECK_SIZEOF(short)dnl
   AC_CHECK_SIZEOF(long, 4)dnl
   AC_CHECK_SIZEOF(char *, 4)dnl
 ])dnl
 
-AC_DEFUN([KDE_DO_IT_ALL],
+AC_DEFUN(KDE_DO_IT_ALL,
 [
 AC_CANONICAL_SYSTEM
 AC_ARG_PROGRAM
@@ -3109,11 +3130,11 @@
 AC_PATH_KDE
 ])
 
-AC_DEFUN([AC_CHECK_RPATH],
+AC_DEFUN(AC_CHECK_RPATH,
 [
 AC_MSG_CHECKING(for rpath)
 AC_ARG_ENABLE(rpath,
-      AC_HELP_STRING([--disable-rpath],[do not use the rpath feature of ld]),
+      [  --disable-rpath         do not use the rpath feature of ld],
       USE_RPATH=$enableval, USE_RPATH=yes)
 
 if test -z "$KDE_RPATH" && test "$USE_RPATH" = "yes"; then
@@ -3139,7 +3160,7 @@
 ])
 
 dnl Check for the type of the third argument of getsockname
-AC_DEFUN([AC_CHECK_SOCKLEN_T], [
+AC_DEFUN(AC_CHECK_SOCKLEN_T, [
   AC_MSG_CHECKING(for socklen_t)
   AC_CACHE_VAL(ac_cv_socklen_t, [
     AC_LANG_SAVE
@@ -3191,7 +3212,7 @@
 dnl >
 dnl for this file it is relicensed under LGPL
 
-AC_DEFUN([AM_KDE_WITH_NLS],
+AC_DEFUN(AM_KDE_WITH_NLS,
   [
     dnl If we use NLS figure out what method
 
@@ -3234,7 +3255,7 @@
 
 dnl AM_PATH_PROG_WITH_TEST_KDE(VARIABLE, PROG-TO-CHECK-FOR,
 dnl   TEST-PERFORMED-ON-FOUND_PROGRAM [, VALUE-IF-NOT-FOUND [, PATH]])
-AC_DEFUN([AM_PATH_PROG_WITH_TEST_KDE],
+AC_DEFUN(AM_PATH_PROG_WITH_TEST_KDE,
 [# Extract the first word of "$2", so it can be a program name with args.
 set dummy $2; ac_word=[$]2
 AC_MSG_CHECKING([for $ac_word])
@@ -3276,7 +3297,7 @@
 
 # serial 1
 
-AC_DEFUN([AM_LC_MESSAGES],
+AC_DEFUN(AM_LC_MESSAGES,
   [if test $ac_cv_header_locale_h = yes; then
     AC_CACHE_CHECK([for LC_MESSAGES], am_cv_val_LC_MESSAGES,
       [AC_TRY_LINK([#include <locale.h>], [return LC_MESSAGES],
@@ -3322,7 +3343,7 @@
 # serial 1
 # Stephan Kulow: I put a KDE in it to avoid name conflicts
 
-AC_DEFUN([AM_KDE_GNU_GETTEXT],
+AC_DEFUN(AM_KDE_GNU_GETTEXT,
   [AC_REQUIRE([AC_PROG_MAKE_SET])dnl
    AC_REQUIRE([AC_PROG_RANLIB])dnl
    AC_REQUIRE([AC_HEADER_STDC])dnl
@@ -3384,14 +3405,14 @@
 
   ])
 
-AC_DEFUN([AC_HAVE_XPM],
+AC_DEFUN(AC_HAVE_XPM,
  [AC_REQUIRE_CPP()dnl
   AC_REQUIRE([KDE_CHECK_EXTRA_LIBS])
 
  test -z "$XPM_LDFLAGS" && XPM_LDFLAGS=
  test -z "$XPM_INCLUDE" && XPM_INCLUDE=
 
- AC_ARG_WITH(xpm,AC_HELP_STRING([--without-xpm],[disable color pixmap XPM tests]),
+ AC_ARG_WITH(xpm, [  --without-xpm           disable color pixmap XPM tests],
 	xpm_test=$withval, xpm_test="yes")
  if test "x$xpm_test" = xno; then
    ac_cv_have_xpm=no
@@ -3439,7 +3460,7 @@
  AC_SUBST(XPMLIB)
 ])
 
-AC_DEFUN([AC_HAVE_DPMS],
+AC_DEFUN(AC_HAVE_DPMS,
  [AC_REQUIRE_CPP()dnl
   AC_REQUIRE([KDE_CHECK_EXTRA_LIBS])
 
@@ -3447,7 +3468,7 @@
  test -z "$DPMS_INCLUDE" && DPMS_INCLUDE=
  DPMS_LIB=
 
- AC_ARG_WITH(dpms,AC_HELP_STRING([--without-dpms],[disable DPMS power saving]),
+ AC_ARG_WITH(dpms, [  --without-dpms          disable DPMS power saving],
 	dpms_test=$withval, dpms_test="yes")
  if test "x$dpms_test" = xno; then
    ac_cv_have_dpms=no
@@ -3540,14 +3561,14 @@
  AC_SUBST(DPMSLIB)
 ])
 
-AC_DEFUN([AC_HAVE_GL],
+AC_DEFUN(AC_HAVE_GL,
  [AC_REQUIRE_CPP()dnl
   AC_REQUIRE([KDE_CHECK_EXTRA_LIBS])
 
  test -z "$GL_LDFLAGS" && GL_LDFLAGS=
  test -z "$GL_INCLUDE" && GL_INCLUDE=
 
- AC_ARG_WITH(gl,AC_HELP_STRING([--without-gl],[disable 3D GL modes]),
+ AC_ARG_WITH(gl, [  --without-gl            disable 3D GL modes],
 	gl_test=$withval, gl_test="yes")
  if test "x$kde_use_qt_emb" = "xyes"; then
    # GL and Qt Embedded is a no-go for now.
@@ -3623,12 +3644,12 @@
 
  dnl shadow password and PAM magic - maintained by ossi@kde.org
 
-AC_DEFUN([KDE_PAM], [
+AC_DEFUN(KDE_PAM, [
   AC_REQUIRE([KDE_CHECK_LIBDL])
 
   want_pam=
   AC_ARG_WITH(pam,
-    AC_HELP_STRING([--with-pam[=ARG]],[enable support for PAM: ARG=[yes|no|service name]]),
+    [  --with-pam[=ARG]        enable support for PAM: ARG=[yes|no|service name]],
     [ if test "x$withval" = "xyes"; then
         want_pam=yes
         pam_service=kde
@@ -3686,9 +3707,9 @@
 ])
 
 dnl DEF_PAM_SERVICE(arg name, full name, define name)
-AC_DEFUN([DEF_PAM_SERVICE], [
+AC_DEFUN(DEF_PAM_SERVICE, [
   AC_ARG_WITH($1-pam,
-    AC_HELP_STRING([--with-$1-pam=[val]],[override PAM service from --with-pam for $2]),
+    [  --with-$1-pam=[val]    override PAM service from --with-pam for $2],
     [ if test "x$use_pam" = xyes; then
         $3_PAM_SERVICE=$withval
       else
@@ -3707,7 +3728,7 @@
     AC_SUBST($3_PAM_SERVICE)
 ])
 
-AC_DEFUN([KDE_SHADOWPASSWD], [
+AC_DEFUN(KDE_SHADOWPASSWD, [
   AC_REQUIRE([KDE_PAM])
 
   AC_CHECK_LIB(shadow, getspent,
@@ -3730,7 +3751,7 @@
   AC_MSG_CHECKING([for shadow passwords])
 
   AC_ARG_WITH(shadow,
-    AC_HELP_STRING([--with-shadow],[If you want shadow password support]),
+    [  --with-shadow		  If you want shadow password support ],
     [ if test "x$withval" != "xno"; then
         use_shadow=yes
       else
@@ -3764,7 +3785,7 @@
 
 ])
 
-AC_DEFUN([KDE_PASSWDLIBS], [
+AC_DEFUN(KDE_PASSWDLIBS, [
   AC_REQUIRE([KDE_MISC_TESTS]) dnl for LIBCRYPT
   AC_REQUIRE([KDE_PAM])
   AC_REQUIRE([KDE_SHADOWPASSWD])
@@ -3790,7 +3811,7 @@
   AC_SUBST(PASSWDLIBS)
 ])
 
-AC_DEFUN([KDE_CHECK_LIBDL],
+AC_DEFUN(KDE_CHECK_LIBDL,
 [
 AC_CHECK_LIB(dl, dlopen, [
 LIBDL="-ldl"
@@ -3805,7 +3826,7 @@
 AC_SUBST(LIBDL)
 ])
 
-AC_DEFUN([KDE_CHECK_DLOPEN],
+AC_DEFUN(KDE_CHECK_DLOPEN,
 [
 KDE_CHECK_LIBDL
 AC_CHECK_HEADERS(dlfcn.h dl.h)
@@ -3820,7 +3841,7 @@
 dnl XXX why change enable_dlopen? its already set by autoconf's AC_ARG_ENABLE
 dnl (MM)
 AC_ARG_ENABLE(dlopen,
-AC_HELP_STRING([--disable-dlopen],[link statically [default=no]]),
+[  --disable-dlopen        link statically [default=no]] ,
 enable_dlopen=$enableval,
 enable_dlopen=yes)
 
@@ -3845,7 +3866,7 @@
 
 ])
 
-AC_DEFUN([KDE_CHECK_DYNAMIC_LOADING],
+AC_DEFUN(KDE_CHECK_DYNAMIC_LOADING,
 [
 KDE_CHECK_DLOPEN(libtool_enable_shared=yes, libtool_enable_static=no)
 KDE_PROG_LIBTOOL
@@ -3865,7 +3886,7 @@
 fi
 ])
 
-AC_DEFUN([KDE_ADD_INCLUDES],
+AC_DEFUN(KDE_ADD_INCLUDES,
 [
 if test -z "$1"; then
   test_include="Pix.h"
@@ -3902,7 +3923,7 @@
 ])
 ])
 
-AC_DEFUN([KDE_CHECK_LIBPTHREAD],
+AC_DEFUN(KDE_CHECK_LIBPTHREAD,
 [
   LIBPTHREAD=""
 
@@ -3920,7 +3941,7 @@
   AC_SUBST(LIBPTHREAD)
 ])
 
-AC_DEFUN([KDE_CHECK_PTHREAD_OPTION],
+AC_DEFUN(KDE_CHECK_PTHREAD_OPTION,
 [
       USE_THREADS=""
       if test -z "$LIBPTHREAD"; then
@@ -3957,7 +3978,7 @@
     AC_SUBST(LIBPTHREAD)
 ])
 
-AC_DEFUN([KDE_CHECK_THREADING],
+AC_DEFUN(KDE_CHECK_THREADING,
 [
   AC_REQUIRE([KDE_CHECK_LIBPTHREAD])
   AC_REQUIRE([KDE_CHECK_PTHREAD_OPTION])
@@ -3971,14 +3992,14 @@
   else
     kde_check_threading_default=yes
   fi
-  AC_ARG_ENABLE(threading,AC_HELP_STRING([--disable-threading],[disables threading even if libpthread found]),
+  AC_ARG_ENABLE(threading, [  --disable-threading     disables threading even if libpthread found ],
    kde_use_threading=$enableval, kde_use_threading=$kde_check_threading_default)
   if test "x$kde_use_threading" = "xyes"; then
     AC_DEFINE(HAVE_LIBPTHREAD, 1, [Define if you have a working libpthread (will enable threaded code)])
   fi
 ])
 
-AC_DEFUN([KDE_TRY_LINK_PYTHON],
+AC_DEFUN(KDE_TRY_LINK_PYTHON,
 [
 if test "$kde_python_link_found" = no; then
 
@@ -4027,7 +4048,7 @@
 
 ])
 
-AC_DEFUN([KDE_CHECK_PYTHON_DIR],
+AC_DEFUN(KDE_CHECK_PYTHON_DIR,
 [
 AC_MSG_CHECKING([for Python directory])
  
@@ -4041,7 +4062,7 @@
 ])
  
 AC_ARG_WITH(pythondir,
-AC_HELP_STRING([--with-pythondir=pythondir],[use python installed in pythondir]),
+[  --with-pythondir=pythondir   use python installed in pythondir ],
 [
   ac_python_dir=$withval
 ], ac_python_dir=$kde_cv_pythondir
@@ -4050,7 +4071,7 @@
 AC_MSG_RESULT($ac_python_dir)
 ])
 
-AC_DEFUN([KDE_CHECK_PYTHON_INTERN],
+AC_DEFUN(KDE_CHECK_PYTHON_INTERN,
 [
 AC_REQUIRE([KDE_CHECK_LIBDL])
 AC_REQUIRE([KDE_CHECK_LIBPTHREAD])
@@ -4142,7 +4163,7 @@
 ])
 
 
-AC_DEFUN([KDE_CHECK_PYTHON],
+AC_DEFUN(KDE_CHECK_PYTHON,
 [
   KDE_CHECK_PYTHON_INTERN("2.3", 
    [KDE_CHECK_PYTHON_INTERN("2.2", 
@@ -4154,7 +4175,7 @@
    ])
 ])
 
-AC_DEFUN([KDE_CHECK_STL],
+AC_DEFUN(KDE_CHECK_STL,
 [
     AC_LANG_SAVE
     AC_LANG_CPLUSPLUS
@@ -4190,7 +4211,7 @@
    AC_LANG_RESTORE
 ])
 
-AC_DEFUN([AC_FIND_QIMGIO],
+AC_DEFUN(AC_FIND_QIMGIO,
    [AC_REQUIRE([AC_FIND_JPEG])
 AC_REQUIRE([KDE_CHECK_EXTRA_LIBS])
 AC_MSG_CHECKING([for qimgio])
@@ -4229,7 +4250,7 @@
 fi
 ])
 
-AC_DEFUN([AM_DISABLE_LIBRARIES],
+AC_DEFUN(AM_DISABLE_LIBRARIES,
 [
     AC_PROVIDE([AM_ENABLE_STATIC])
     AC_PROVIDE([AM_ENABLE_SHARED])
@@ -4238,7 +4259,7 @@
 ])
 
 
-AC_DEFUN([AC_CHECK_UTMP_FILE],
+AC_DEFUN(AC_CHECK_UTMP_FILE,
 [
     AC_MSG_CHECKING([for utmp file])
 
@@ -4271,7 +4292,7 @@
 ])
 
 
-AC_DEFUN([KDE_CREATE_SUBDIRSLIST],
+AC_DEFUN(KDE_CREATE_SUBDIRSLIST,
 [
 
 DO_NOT_COMPILE="$DO_NOT_COMPILE CVS debian bsd-port admin"
@@ -4318,7 +4339,7 @@
 AC_SUBST(TOPSUBDIRS)
 ])
 
-AC_DEFUN([KDE_CHECK_NAMESPACES],
+AC_DEFUN(KDE_CHECK_NAMESPACES,
 [
 AC_MSG_CHECKING(whether C++ compiler supports namespaces)
 AC_LANG_SAVE
@@ -4348,7 +4369,7 @@
 dnl Check for S_ISSOCK macro. Doesn't exist on Unix SCO. faure@kde.org
 dnl ------------------------------------------------------------------------
 dnl
-AC_DEFUN([AC_CHECK_S_ISSOCK],
+AC_DEFUN(AC_CHECK_S_ISSOCK,
 [
 AC_MSG_CHECKING(for S_ISSOCK)
 AC_CACHE_VAL(ac_cv_have_s_issock,
@@ -4383,7 +4404,7 @@
 dnl Check for MAXPATHLEN macro, defines KDEMAXPATHLEN. faure@kde.org
 dnl ------------------------------------------------------------------------
 dnl
-AC_DEFUN([AC_CHECK_KDEMAXPATHLEN],
+AC_DEFUN(AC_CHECK_KDEMAXPATHLEN,
 [
 AC_MSG_CHECKING(for MAXPATHLEN)
 AC_CACHE_VAL(ac_cv_maxpathlen,
@@ -4417,7 +4438,7 @@
 AC_DEFINE_UNQUOTED(KDEMAXPATHLEN,$ac_cv_maxpathlen, [Define a safe value for MAXPATHLEN] )
 ])
 
-AC_DEFUN([KDE_CHECK_HEADER],
+AC_DEFUN(KDE_CHECK_HEADER,
 [
    AC_LANG_SAVE
    kde_safe_cppflags=$CPPFLAGS
@@ -4428,7 +4449,7 @@
    AC_LANG_RESTORE
 ])
 
-AC_DEFUN([KDE_CHECK_HEADERS],
+AC_DEFUN(KDE_CHECK_HEADERS,
 [
    AH_CHECK_HEADERS([$1])
    AC_LANG_SAVE
@@ -4440,14 +4461,14 @@
    AC_LANG_RESTORE
 ])
 
-AC_DEFUN([KDE_FAST_CONFIGURE],
+AC_DEFUN(KDE_FAST_CONFIGURE,
 [
   dnl makes configure fast (needs perl)
-  AC_ARG_ENABLE(fast-perl, AC_HELP_STRING([--disable-fast-perl],[disable fast Makefile generation (needs perl)]),
+  AC_ARG_ENABLE(fast-perl, [  --disable-fast-perl     disable fast Makefile generation (needs perl)],
       with_fast_perl=$enableval, with_fast_perl=yes)
 ])
 
-AC_DEFUN([KDE_CONF_FILES],
+AC_DEFUN(KDE_CONF_FILES,
 [
   val=
   if test -f $srcdir/configure.files ; then
@@ -4462,7 +4483,7 @@
   AC_SUBST(CONF_FILES)
 ])dnl
 
-AC_DEFUN([KDE_SET_PREFIX],
+AC_DEFUN(KDE_SET_PREFIX,
 [
   unset CDPATH
   dnl make $KDEDIR the default for the installation
@@ -4545,7 +4566,7 @@
   fi
 ])dnl
 
-AC_DEFUN([KDE_LANG_CPLUSPLUS],
+AC_DEFUN(KDE_LANG_CPLUSPLUS,
 [AC_LANG_CPLUSPLUS
 ac_link='rm -rf SunWS_cache; ${CXX-g++} -o conftest${ac_exeext} $CXXFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS 1>&AC_FD_CC'
 pushdef([AC_LANG_CPLUSPLUS], [popdef([AC_LANG_CPLUSPLUS]) KDE_LANG_CPLUSPLUS])
@@ -4556,7 +4577,7 @@
 KDE_LANG_CPLUSPLUS
 ])
 
-AC_DEFUN([KDE_CHECK_LONG_LONG],
+AC_DEFUN(KDE_CHECK_LONG_LONG,
 [
 AC_MSG_CHECKING(for long long)
 AC_CACHE_VAL(kde_cv_c_long_long,
@@ -4576,7 +4597,7 @@
 fi
 ])
 
-AC_DEFUN([KDE_CHECK_LIB],
+AC_DEFUN(KDE_CHECK_LIB,
 [
      kde_save_LDFLAGS="$LDFLAGS"
      dnl AC_CHECK_LIB modifies LIBS, so save it here
@@ -4592,7 +4613,7 @@
      LIBS="$kde_save_LIBS"
 ])
 
-AC_DEFUN([KDE_JAVA_PREFIX],
+AC_DEFUN(KDE_JAVA_PREFIX,
 [
 	dir=`dirname "$1"`
 	base=`basename "$1"`
@@ -4615,11 +4636,11 @@
 ])
 
 dnl KDE_CHEC_JAVA_DIR(onlyjre)
-AC_DEFUN([KDE_CHECK_JAVA_DIR],
+AC_DEFUN(KDE_CHECK_JAVA_DIR,
 [
 
 AC_ARG_WITH(java,
-AC_HELP_STRING([--with-java=javadir],[use java installed in javadir, --without-java disables]),
+[  --with-java=javadir     use java installed in javadir, --without-java disables ],
 [  ac_java_dir=$withval
 ], ac_java_dir=""
 )
@@ -4865,7 +4886,7 @@
 m4_define([AC_FOREACH],
 [mm_foreach([$1], m4_split(m4_normalize([$2])), [$3])])
 
-AC_DEFUN([KDE_NEED_FLEX],
+AC_DEFUN(KDE_NEED_FLEX,
 [
 kde_libs_safe=$LIBS
 LIBS="$LIBS $USER_LDFLAGS"
@@ -4877,7 +4898,7 @@
 AC_SUBST(LEXLIB)
 ])
 
-AC_DEFUN([AC_PATH_QTOPIA],
+AC_DEFUN(AC_PATH_QTOPIA,
 [
   dnl TODO: use AC_CACHE_VAL
 
@@ -4906,7 +4927,7 @@
   ac_qtopia_incdir=NO
 
   AC_ARG_WITH(qtopia-dir,
-              AC_HELP_STRING([--with-qtopia-dir=DIR],[where the root of Qtopia is installed]),
+              [  --with-qtopia-dir=DIR   where the root of Qtopia is installed ],
               [  ac_qtopia_incdir="$withval"/include] ) 
   
   qtopia_incdirs=""
@@ -4981,7 +5002,7 @@
 ])
 
 
-AC_DEFUN([KDE_INIT_DOXYGEN],
+AC_DEFUN(KDE_INIT_DOXYGEN,
 [
 AC_MSG_CHECKING([for Qt docs])
 kde_qtdir=
@@ -5018,7 +5039,7 @@
 ])
 
 
-AC_DEFUN([AC_FIND_BZIP2],
+AC_DEFUN(AC_FIND_BZIP2,
 [
 AC_MSG_CHECKING([for bzDecompress in libbz2])
 AC_CACHE_VAL(ac_cv_lib_bzip2,
@@ -5105,7 +5126,7 @@
 dnl and $(SSL_INCLUDES) will be -Isslhdrlocation (if needed)
 dnl ------------------------------------------------------------------------
 dnl
-AC_DEFUN([KDE_CHECK_SSL],
+AC_DEFUN(KDE_CHECK_SSL,
 [
 LIBSSL="-lssl -lcrypto"
 AC_REQUIRE([KDE_CHECK_LIB64])
@@ -5114,14 +5135,14 @@
 ssl_libraries=""
 ssl_includes=""
 AC_ARG_WITH(ssl-dir,
-    AC_HELP_STRING([--with-ssl-dir=DIR],[where the root of OpenSSL is installed]),
+    [  --with-ssl-dir=DIR      where the root of OpenSSL is installed],
     [  ac_ssl_includes="$withval"/include
        ac_ssl_libraries="$withval"/lib$kdelibsuff
     ])
 
 want_ssl=yes
 AC_ARG_WITH(ssl,
-    AC_HELP_STRING([--without-ssl],[disable SSL checks]),
+    [  --without-ssl           disable SSL checks],
     [want_ssl=$withval])
 
 if test $want_ssl = yes; then
@@ -5276,7 +5297,7 @@
   SSL_INCLUDES="-I$ssl_includes"
 fi
 
-if test "$ssl_libraries" = "/usr/lib" || test "$ssl_libraries" = "/usr/local/lib" || test -z "$ssl_libraries" || test "$ssl_libraries" = "NONE"; then
+if test "$ssl_libraries" = "/usr/lib" || test "$ssl_libraries" = "/usr/local/lib" || test -z "$ssl_libraries"; then
  SSL_LDFLAGS=""
 else
  SSL_LDFLAGS="-L$ssl_libraries -R$ssl_libraries"
@@ -5287,7 +5308,7 @@
 AC_SUBST(LIBSSL)
 ])
 
-AC_DEFUN([KDE_CHECK_STRLCPY],
+AC_DEFUN(KDE_CHECK_STRLCPY,
 [
   AC_REQUIRE([AC_CHECK_STRLCAT])
   AC_REQUIRE([AC_CHECK_STRLCPY])
@@ -5308,7 +5329,7 @@
   ])
 ])
 
-AC_DEFUN([KDE_CHECK_BINUTILS],
+AC_DEFUN(KDE_CHECK_BINUTILS,
 [
   AC_MSG_CHECKING([if ld supports unversioned version maps])
 
@@ -5343,7 +5364,7 @@
 _AM_IF_OPTION([no-dependencies],, [_AM_DEPENDENCIES(OBJC)])
 ])
 
-AC_DEFUN([KDE_CHECK_PERL],
+AC_DEFUN(KDE_CHECK_PERL,
 [
 	KDE_FIND_PATH(perl, PERL, [$bindir $exec_prefix/bin $prefix/bin], [
 		    AC_MSG_ERROR([No Perl found in your $PATH.
Index: am_edit
===================================================================
RCS file: /home/kde/kde-common/admin/am_edit,v
retrieving revision 1.409
retrieving revision 1.406
diff -u -r1.409 -r1.406
--- admin/am_edit~	29 Jan 2004 14:58:27 -0000	1.409
+++ admin/am_edit	29 Oct 2003 22:45:50 -0000	1.406
@@ -269,7 +269,7 @@
 	tag_CLOSURE();
 	tag_NMCHECK();
 	tag_UIFILES();              # Sorts out ui rules
-	tag_KCFGFILES();            # Sorts out kcfg rules
+	tag_KCFGFILES();            # Sourt out kcfg rules
         tag_METASOURCES ();         # Sorts out the moc rules
         if ($sources_changed{$program}) {
             my $lookup = $program . '_SOURCES\s*=[ \t]*(.*)';
@@ -892,7 +892,7 @@
                     substituteLine($lookup, "${kdeinit}_la_LIBADD = libkdeinit_${kdeinit}.la");
                     appendLines("libkdeinit_${kdeinit}_la_LIBADD = $libadd\n");
                 }
-                appendLines("libkdeinit_${kdeinit}_la_LDFLAGS = -no-undefined -avoid-version \$(all_libraries)\n");
+                appendLines("libkdeinit_${kdeinit}_la_LDFLAGS = -avoid-version \$(all_libraries)\n");
 
                 # add library dependencies
                 $lookup = $kdeinit . '_la_DEPENDENCIES\s*=[ \t]*(.*)';
@@ -1360,8 +1360,7 @@
                 my $dep_lines = "$source.$cxxsuffix: $sourcedir$source.ui $source.h $source.moc\n";
                 $dep_lines .= "\trm -f $source.$cxxsuffix\n";
                 if (!$kdeopts{"qtonly"}) {
-                    $dep_lines .= "\techo '#include <kdialog.h>' > $source.$cxxsuffix\n";
-                    $dep_lines .= "\techo '#include <klocale.h>' >> $source.$cxxsuffix\n";
+                    $dep_lines .= "\techo '#include <klocale.h>' > $source.$cxxsuffix\n";
                     my ($mangled_source) = $source;
                     $mangled_source =~ s/[^A-Za-z0-9]/_/g;  # get rid of garbage
                     $dep_lines .= "\t\$(UIC) -tr \${UIC_TR} -i $source.h $sourcedir$source.ui > $source.$cxxsuffix.temp ; ret=\$\$?; \\\n";
@@ -1426,14 +1425,10 @@
                 $kcfg = "$program.kcfg";
                 findKcfgFile("$source.kcfgc");
 
-                my $fixsuffix = "";
-                $fixsuffix = "else mv $source.cpp $source.$cxxsuffix ; " 
-                    unless "cpp" eq $cxxsuffix;
-
                 my $dep_lines = "$source.$cxxsuffix: $source.h\n";
                 $dep_lines .= "$source.h: $sourcedir$kcfg $sourcedir$source.kcfgc \$(KCFG_DEPENDENCIES)\n";
                 $dep_lines .= "\t\$(KCONFIG_COMPILER) $sourcedir$kcfg $sourcedir$source.kcfgc; ret=\$\$?; \\\n";
-		$dep_lines .= "\tif test \"\$\$ret\" != 0; then rm -f $source.h ; exit \$\$ret ; $fixsuffix fi\n\n";
+		$dep_lines .= "\tif test \"\$\$ret\" != 0; then rm -f $source.h ; exit \$\$ret ; fi\n\n";
 
                 $rule_adds{"$source.$cxxsuffix"} = $dep_lines;
 
@@ -2347,7 +2342,7 @@
     $MakefileData =~ s/\034/\\\n/g;    # Restore continuation lines
     # Append our $progId line, _below_ the "generated by automake" line
     # because automake-1.6 relies on the first line to be his own.
-    my $progIdLine = "\# $progId - " . '$Revision$ '."\n";
+    my $progIdLine = "\# $progId - " . '$Revision$ '."\n";
     if ( !( $MakefileData =~ s/^(.*generated .*by automake.*\n)/$1$progIdLine/ ) ) {
         warn "automake line not found in $makefile\n";
 	# Fallback: first line
Index: config.pl
===================================================================
RCS file: /home/kde/kde-common/admin/config.pl,v
retrieving revision 1.20
retrieving revision 1.19
diff -u -r1.20 -r1.19
--- admin/config.pl~	26 Dec 2003 17:16:14 -0000	1.20
+++ admin/config.pl	19 Nov 2003 10:21:14 -0000	1.19
@@ -30,8 +30,6 @@
 
 use strict;
 
-use File::Path;
-
 my $ac_subs=$ARGV[0];
 my $ac_sacfiles = $ARGV[1];
 my $ac_given_srcdir=$ARGV[2];
@@ -139,7 +137,7 @@
     ($ac_dir = $ac_file) =~ s%/[^/][^/]*$%%;
     if ( ($ac_dir ne $ac_file) && ($ac_dir ne ".")) {
 # The file is in a subdirectory.
-	if (! -d "$ac_dir") { mkpath "$ac_dir", 0, 0777; }
+	if (! -d "$ac_dir") { mkdir "$ac_dir", 0777; }
 	($ac_dir_suffix = $ac_dir) =~ s%^./%%;
 	$ac_dir_suffix="/".$ac_dir_suffix;
 # A "../" for each directory in $ac_dir_suffix.
Index: cvs.sh
===================================================================
RCS file: /home/kde/kde-common/admin/cvs.sh,v
retrieving revision 1.120
retrieving revision 1.115
diff -u -r1.120 -r1.115
--- admin/cvs.sh~	30 Jan 2004 09:11:51 -0000	1.120
+++ admin/cvs.sh	3 Dec 2003 12:40:42 -0000	1.115
@@ -27,18 +27,17 @@
 
 check_autotool_versions()
 {
-required_autoconf_version="2.53 or newer"
 AUTOCONF_VERSION=`$AUTOCONF --version | head -n 1`
 case $AUTOCONF_VERSION in
   Autoconf*2.5* | autoconf*2.5* ) : ;;
   "" )
     echo "*** AUTOCONF NOT FOUND!."
-    echo "*** KDE requires autoconf $required_autoconf_version"
+    echo "*** KDE requires autoconf >= 2.52"
     exit 1
     ;;
   * )
     echo "*** YOU'RE USING $AUTOCONF_VERSION."
-    echo "*** KDE requires autoconf $required_autoconf_version"
+    echo "*** KDE requires autoconf >= 2.52"
     exit 1
     ;;
 esac
@@ -48,12 +47,12 @@
   Autoconf*2.5* | autoheader*2.5* ) : ;;
   "" )
     echo "*** AUTOHEADER NOT FOUND!."
-    echo "*** KDE requires autoheader $required_autoconf_version"
+    echo "*** KDE requires autoheader >= 2.52 (part of autoconf)"
     exit 1
     ;;
   * )
     echo "*** YOU'RE USING $AUTOHEADER_VERSION."
-    echo "*** KDE requires autoheader $required_autoconf_version"
+    echo "*** KDE requires autoheader >= 2.52 (part of autoconf)"
     exit 1
     ;;
 esac
@@ -66,7 +65,7 @@
     echo "*** KDE requires automake $required_automake_version"
     exit 1
     ;;
-  automake*1.6.* | automake*1.7* | automake*1.8* ) : ;;
+  automake*1.6.* | automake*1.7* ) : ;;
   "" )
     echo "*** AUTOMAKE NOT FOUND!."
     echo "*** KDE requires automake $required_automake_version"
@@ -309,7 +308,7 @@
    fi
 fi
 if test -z "$VERSION" || test "$VERSION" = "@VERSION@"; then
-     VERSION="\"3.2.0\""
+     VERSION="\"3.1.94\""
 fi
 if test -z "$modulename" || test "$modulename" = "@MODULENAME@"; then
    modulename=`pwd`; 
@@ -360,10 +359,6 @@
 fi; done
 test -f configure.in.mid && echo configure.in.mid >> configure.files
 test -f configure.in.bot && echo configure.in.bot >> configure.files
-if test ! -s configure.files; then
-   echo "There are no files to build a configure. Please check your checkout."
-   exit 1
-fi
 }
 
 create_subdirs()
@@ -601,12 +596,12 @@
 rm -rf po.backup
 }
 
-unset LC_ALL || :
-unset LANG || :
-unset LC_CTYPE || :
-unset LANGUAGE || :
+unset LC_ALL
+unset LANG
+unset LC_CTYPE
+unset LANGUAGE
 
-unset CDPATH || :
+unset CDPATH
 admindir=`echo "$0" | sed 's%[\\/][^\\/][^\\/]*$%%'`
 test "x$admindir" = "x$0" && admindir=.
 
Index: debianrules
===================================================================
RCS file: /home/kde/kde-common/admin/debianrules,v
retrieving revision 1.115
retrieving revision 1.114
diff -u -r1.115 -r1.114
--- admin/debianrules~	7 Jan 2004 17:08:43 -0000	1.115
+++ admin/debianrules	7 Sep 2003 15:37:46 -0000	1.114
@@ -40,7 +40,7 @@
 
 if (defined $ENV{DEB_BUILD_OPTIONS} &&
     $ENV{DEB_BUILD_OPTIONS} =~ /\bdebug\b/) {
-  $enable_debug="--enable-debug=full";
+  $enable_debug="--enable-debug";
   $enable_final="";
 }
 else {
Index: oldinclude.m4.in
===================================================================
RCS file: /home/kde/kde-common/admin/oldinclude.m4.in,v
retrieving revision 2.2
retrieving revision 2.1
diff -u -r2.2 -r2.1
--- admin/oldinclude.m4.in~	7 Jan 2004 01:15:32 -0000	2.2
+++ admin/oldinclude.m4.in	13 Sep 2003 10:52:09 -0000	2.1
@@ -19,7 +19,7 @@
 dnl    the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 dnl    Boston, MA 02111-1307, USA.
 
-AC_DEFUN([KDE_CHECK_MICO],
+AC_DEFUN(KDE_CHECK_MICO,
 [
 AC_REQUIRE([KDE_CHECK_LIBDL])
 AC_REQUIRE([KDE_MISC_TESTS])
@@ -128,7 +128,7 @@
 
 ])
 
-AC_DEFUN([KDE_CHECK_MINI_STL],
+AC_DEFUN(KDE_CHECK_MINI_STL,
 [
 AC_REQUIRE([KDE_CHECK_MICO])
 
@@ -165,11 +165,11 @@
 
 ])
 
-AC_DEFUN([KDE_CHECK_ANSI],
+AC_DEFUN(KDE_CHECK_ANSI,
 [
 ])
 
-AC_DEFUN([KDE_CHECK_INSURE],
+AC_DEFUN(KDE_CHECK_INSURE,
 [
   AC_ARG_ENABLE(insure, [  --enable-insure             use insure++ for debugging [default=no]],
   [
@@ -186,7 +186,7 @@
    fi
 ])
 
-AC_DEFUN([KDE_CHECK_NEWLIBS],
+AC_DEFUN(KDE_CHECK_NEWLIBS,
 [
 
-])
+])
\ No newline at end of file
