cvs server: Diffing .
Index: sidebar.cc
===================================================================
RCS file: /home/kde/koffice/kpresenter/sidebar.cc,v
retrieving revision 1.9
diff -u -p -r1.9 sidebar.cc
--- koffice/kpresenter/sidebar.cc	2001/07/03 09:52:50	1.9
+++ koffice/kpresenter/sidebar.cc	2002/02/22 14:09:21
@@ -172,9 +172,12 @@ void SideBar::rightButtonPressed( QListV
 
 void SideBar::renamePageTitle()
 {
-    int pageNumber = QListView::selectedItem()->text( 1 ).toInt() - 1;
+  QListViewItem *item=QListView::selectedItem();
+  if(!item)
+    return;
+    int pageNumber = item->text( 1 ).toInt() - 1;
     bool ok;
-    QString activeTitle = QListView::selectedItem()->text( 0 );
+    QString activeTitle = item->text( 0 );
     QString newTitle = KLineEditDlg::getText( i18n("Page Title"), activeTitle, &ok, this );
 
     // Have a different name ?
cvs server: Diffing autoformEdit
cvs server: Diffing autoforms
cvs server: Diffing autoforms/Arrows
cvs server: Diffing autoforms/Connections
cvs server: Diffing dtd
cvs server: Diffing pics
cvs server: Diffing scripts
cvs server: Diffing slideshow
cvs server: Diffing templates
cvs server: Diffing templates/A4
cvs server: Diffing templates/Screen
cvs server: Diffing templates/Screenpresentations
cvs server: Diffing templates/common_desktop
cvs server: Diffing templates/common_icon
cvs server: Diffing templates/legal
cvs server: Diffing templates/letter
cvs server: Diffing toolbar
