--- kexi/kexidb/drivers/pqxx/pqxxconnection.cpp.orig	Sun Feb 11 16:58:22 2007
+++ kexi/kexidb/drivers/pqxx/pqxxconnection.cpp	Sun Feb 11 16:58:53 2007
@@ -277,7 +277,7 @@
 		//		m_trans = new pqxx::nontransaction(*m_pqxxsql);
 //		KexiDBDrvDbg << "About to execute" << endl;
 		//Create a result object through the transaction
-		d->res = new pqxx::result(m_trans->data->exec(statement.utf8()));
+		d->res = new pqxx::result(m_trans->data->exec(statement.utf8().data()));
 //		KexiDBDrvDbg << "Executed" << endl;
 		//Commit the transaction
 		if (implicityStarted) {
--- kexi/kexidb/drivers/pqxx/pqxxconnection_p.h.orig	Sun Feb 11 16:52:10 2007
+++ kexi/kexidb/drivers/pqxx/pqxxconnection_p.h	Sun Feb 11 16:54:21 2007
@@ -31,7 +31,7 @@
 #define PQXXSQLCONNECTIONINTERNAL_H
 
 #include <kexidb/connection_p.h>
-#include <pqxx/all.h>
+#include <pqxx/pqxx>
 
 namespace KexiDB
 {
--- kexi/kexidb/drivers/pqxx/pqxxcursor.cpp.orig	Sun Feb 11 16:45:47 2007
+++ kexi/kexidb/drivers/pqxx/pqxxcursor.cpp	Sun Feb 11 16:50:26 2007
@@ -155,7 +155,7 @@
 			m_implicityStarted = true;
 		}
 
-		m_res = new pqxx::result(((pqxxSqlConnection*)connection())->m_trans->data->exec(m_sql.utf8()));
+		m_res = new pqxx::result(((pqxxSqlConnection*)connection())->m_trans->data->exec(m_sql.utf8().data()));
 		((pqxxSqlConnection*)connection())
 			->drv_commitTransaction(((pqxxSqlConnection*)connection())->m_trans);
 //		my_conn->m_trans->commit();
--- kexi/kexidb/drivers/pqxx/pqxxdriver.cpp.orig	Sun Feb 11 16:38:08 2007
+++ kexi/kexidb/drivers/pqxx/pqxxdriver.cpp	Sun Feb 11 17:29:18 2007
@@ -133,14 +133,14 @@
 //
 QString pqxxSqlDriver::escapeString( const QString& str) const
 {
-    return QString(pqxx::Quote(str.ascii()).c_str());
+    return "'"+QString(pqxx::sqlesc(str.ascii()).c_str())+"'";
 }
 
 //==================================================================================
 //
 QCString pqxxSqlDriver::escapeString( const QCString& str) const
 {
-    return QCString(pqxx::Quote(QString(str).ascii()).c_str());
+    return "'"+QCString(pqxx::sqlesc(QString(str).ascii()).c_str())+"'";
 }
 
 //==================================================================================
--- kexi/migration/pqxx/pqxxmigrate.h.orig	Sun Feb 11 17:15:12 2007
+++ kexi/migration/pqxx/pqxxmigrate.h	Sun Feb 11 17:15:38 2007
@@ -26,7 +26,7 @@
 #include <kexidb/field.h>
 #include <kexidb/connection.h>
 
-#include <pqxx/all.h>
+#include <pqxx/pqxx>
 
 namespace KexiMigration
 {
