--- koffice-040509/krita/core/kis_image.cc.orig	2004-05-10 00:03:51.000000000 +0200
+++ koffice-040509/krita/core/kis_image.cc	2004-05-10 11:33:30.218796504 +0200
@@ -506,12 +506,12 @@
 
 const KisLayerSP KisImage::activeLayer() const
 {
-	return m_selection ? m_selection : m_activeLayer;
+	return m_selection ? static_cast<KisLayerSP>(m_selection) : m_activeLayer;
 }
 
 KisLayerSP KisImage::activeLayer()
 {
-	return m_selection ? m_selection : m_activeLayer;
+	return m_selection ? static_cast<KisLayerSP>(m_selection) : m_activeLayer;
 }
 
 KisLayerSP KisImage::activate(KisLayerSP layer)
