--- koffice-031105/kexi/plugins/tables/Makefile.am.orig	2003-09-30 01:01:28.000000000 +0200
+++ koffice-031105/kexi/plugins/tables/Makefile.am	2003-11-05 13:35:16.000000000 +0100
@@ -1,6 +1,6 @@
 kde_module_LTLIBRARIES = kexihandler_table.la
 
-kexihandler_table_la_SOURCES = kexitablepart.cpp kexitablepartproxy.cpp #kexialtertable.cpp
+kexihandler_table_la_SOURCES = kexitablepart.cpp kexitablepartproxy.cpp
 kexihandler_table_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
 kexihandler_table_la_LIBADD = ../../core/libkexicore.la ../../tableview/libkexidatatable.la \
 		../../widget/libkexiextendedwidgets.la ../../kexidb/libkexidb.la 
