--- koffice-031024/kexi/kexidb/drivers/configure.in.in.orig	2003-10-09 01:59:54.000000000 +0200
+++ koffice-031024/kexi/kexidb/drivers/configure.in.in	2003-10-28 08:14:48.000000000 +0100
@@ -38,7 +38,7 @@
 ])
 
 if test -n "$MYSQL_CONFIG"; then
-	mysql_incdir=`$MYSQL_CONFIG --cflags| $SED -e "s,-I',,g" | $SED -e "s,'$,,g"`
+	mysql_incdir=`$MYSQL_CONFIG --cflags| $SED -e "s,-I,,g" -e "s,',,g"`
 	mysql_libdir=`$MYSQL_CONFIG --libs| $SED -e "s,',,g"`
 	MYSQL_INC=$mysql_incdir
 	MYSQL_LIBS=$mysql_libdir
