--- koffice-031113/kexi/widget/Makefile.am.orig	2003-11-11 03:12:30.000000000 +0100
+++ koffice-031113/kexi/widget/Makefile.am	2003-11-13 13:42:21.000000000 +0100
@@ -1,7 +1,7 @@
 include $(top_srcdir)/kexi/Makefile.global
 
 lib_LTLIBRARIES = libkexiextendedwidgets.la
-libkexiextendedwidgets_la_SOURCES = kexidatatable.cpp kexidatatableview.cpp #kexirecordnavigator.cpp kexifilterdlg.cpp kexiquerydesignersqleditor.cpp
+libkexiextendedwidgets_la_SOURCES = kexidatatable.cpp kexidatatableview.cpp
 libkexiextendedwidgets_la_LDFLAGS = $(all_libraries) -Wnounresolved
 libkexiextendedwidgets_la_LIBADD = $(LIB_KDEUI) ../tableview/libkexidatatable.la ../core/libkexicore.la
 
